### PR TITLE
feat(api) new endpoint DeleteService

### PR DESCRIPTION
**Pull request by a-launois**
## Description

new configuration endpoint 
```
DELETE /configuration/services/{serviceId}
```

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Core/Service/Application/UseCase/DeleteService/DeleteService.php [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            }
+
+            if ($this->user->isAdmin()) {
+                $isServiceValid = $this->readRepository->exists($serviceId);
+            } else {
+                $accessGroups = $this->readAccessGroupRepository->findByContact($this->user);
+                $isServiceValid = $this->readRepository->existsByAccessGroups($serviceId, $accessGroups);
+            }
+
+            if ($isServiceValid === false) {
```
Comment (abuathier) [inline centreon/src/Core/Service/Application/UseCase/DeleteService/DeleteService.php:88]: Something more explicit maybe ?
```suggestion
                $doesServiceExists = $this->readRepository->exists($serviceId);
            } else {
                $accessGroups = $this->readAccessGroupRepository->findByContact($this->user);
                $doesServiceExists = $this->readRepository->existsByAccessGroups($serviceId, $accessGroups);
            }

            if ($doesServiceExists) {
```

File: centreon/src/Core/Service/Application/UseCase/DeleteService/DeleteService.php [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+                return;
+            }
+
+            try {
+                $monitoringServerId = $this->readRepository->findMonitoringServerId($serviceId);
+                $this->writeRepository->delete($serviceId);
+                $this->writeMonitoringServerRepository->notifyConfigurationChange($monitoringServerId);
+
+                $this->storageEngine->commitTransaction();
```
Comment (abuathier) [inline centreon/src/Core/Service/Application/UseCase/DeleteService/DeleteService.php:100]: Missing `beginTransaction` somewhere if I am not wrong.

**Review by cg-tw — DISMISSED** [review]

File: centreon/tests/api/features/ServiceOnPrem.feature
Comment (callapa) [inline centreon/tests/api/features/ServiceOnPrem.feature:1]: You've forgotten to test API access rights (HTTP code 403).

File: centreon/src/Core/Service/Infrastructure/Repository/DbWriteServiceRepository.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        $this->db = $db;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function delete(int $serviceId): void
+    {
+        $this->info('Delete service by ID');
```
Comment (callapa) [inline centreon/src/Core/Service/Infrastructure/Repository/DbWriteServiceRepository.php:48]: Please move the logs to the use cases side.

File: centreon/src/Core/Service/Infrastructure/Repository/DbReadServiceRepository.php [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+            ->appendWhere(
+                <<<'SQL'
+                    WHERE s.service_id = :service_id
+                    SQL
+            )
+            ->appendWhere(
+                <<<'SQL'
+                    WHERE s.service_register = '1'
+                    SQL
+            )
```
Comment (callapa) [inline centreon/src/Core/Service/Infrastructure/Repository/DbReadServiceRepository.php:136]: More understandable than two concatenated WHERE conditions, don't you think?
```suggestion
            ->appendWhere(
                <<<'SQL'
                    WHERE s.service_id = :service_id AND s.service_register = '1'
                    SQL
            )
```

**Review by callapa — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Pull request merged**